chat-ui/src/services/conversations.ts

```diff
@@ -292,14 +291,23 @@ export class ConversationService {
     conversationId: string;
     messageId: string;
     rating: boolean;
-  }): Promise<void> {
+  }): Promise<boolean> {
     const path = `/conversations/${conversationId}/messages/${messageId}/rating`;
-    await fetch(this.getUrl(path), {
+    const res = await fetch(this.getUrl(path), {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
       },
       body: JSON.stringify({ rating }),
     });
+
+    if(res.status === 204) {
+      return rating;
+    }
+    if (res.status >= 400) {
+      const data = await res.json();
+      throw new Error(data.error);
+    }
+    throw new Error(`Server error: ${res.statusText ?? res.status}`);
```

Comment:
Is it necessarily an error here? We could say
```suggestion
    throw new Error(`Server returned unexpected status ${res.status} (message: ${res.statusText})`);
```
(Will render as 'undefined' for undefined status)

Reply:
5XX errors _do_ imply some kind of server error but the error message doesn't necessarily need to say that verbatim